service/controller/resource/workermigration/resource.go

```diff
@@ -0,0 +1,50 @@
+package workermigration
+
+import (
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	azureclient "github.com/giantswarm/azure-operator/v4/client"
+	"github.com/giantswarm/azure-operator/v4/service/controller/resource/workermigration/internal/azure"
+)
+
+const (
+	Name = "workermigration"
+)
+
+type Config struct {
+	ClientFactory *azureclient.Factory
+	CtrlClient    client.Client
+	Logger        micrologger.Logger
+}
+
+type Resource struct {
+	azureapi   azure.API
+	ctrlClient client.Client
+	logger     micrologger.Logger
+}
+
+func New(config Config) (*Resource, error) {
+	if config.ClientFactory == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.ClientFactory must not be empty", config)
+	}
+	if config.CtrlClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.CtrlClient must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	newResource := &Resource{
+		azureapi:   azure.NewAPI(config.ClientFactory),
```

Comment:
Since the abstracted API implementation is still under `internal` I'm hiding it here. Doesn't affect testing since it's in the same package.